src/sortStudents.ts

```diff
@@ -1,16 +1,53 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+) :
+  Student[] {
+  const avarageGrade = (grades: number[]): number => {
+    return grades.reduce((sum, grade) => sum + grade) / grades.length;
+  };
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return [...students].sort((studentA, studentB) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? studentA[sortBy].localeCompare(studentB[sortBy])
+          : studentB[sortBy].localeCompare(studentA[sortBy]);
+
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? +(studentA[sortBy]) - +(studentB[sortBy])
+          : +(studentB[sortBy]) - +(studentA[sortBy]);
```

Comment:
Аvoid duality of characters such as `- +` 
```suggestion
          ? +(studentA[sortBy]) - Number(studentB[sortBy])
          : +(studentB[sortBy]) - Number(studentA[sortBy]);
```